src/include/daos_fs.h

```diff
@@ -119,6 +119,19 @@ dfs_open(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 	 int flags, daos_oclass_id_t cid, daos_size_t chunk_size,
 	 const char *value, dfs_obj_t **obj);
 
+/*
+ * Duplicate the DFS object without any RPCs (locally) by using the existing
+ * open handles. This is used mostly for low-level fuse to avoid re-opening. The
+ * duplicated object must be released with dfs_release().
+ *
+ * \param[in]	obj	Object to dup.
+ * \param[out]	new_obj	DFS object that is duplicated/opened.
+ *
+ * \return		0 on Success. Negative on Failure.
+ */
+int
+dfs_dup(dfs_t *dfs, dfs_obj_t *obj, dfs_obj_t **new_obj);
```

Comment:
This needs to take a mode option.

Reply:
mode or just a flag for RO, RW, WO?

Reply:
i would think the mode should be inherited from the source object mode

Reply:
That doesn't work, if lookup() opens everything with RW then you won't be able to open RO files, and if lookup opens everything with RO then you won't be able to write anything.